src/main/java/frc/robot/subsystems/vision/VisionInterface.java

```diff
@@ -0,0 +1,57 @@
+package frc.robot.subsystems.vision;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import frc.robot.extras.vision.MegatagPoseEstimate;
+import frc.robot.subsystems.vision.VisionConstants.Limelight;
+import org.littletonrobotics.junction.AutoLog;
+
+public interface VisionInterface {
+  @AutoLog
+  class VisionInputs {
+    public boolean[] isLimelightConnected = new boolean[Limelight.values().length];
+
+    public MegatagPoseEstimate[] limelightMegatagPose =
+        new MegatagPoseEstimate[Limelight.values().length];
+    public double[] limelightLatency = new double[Limelight.values().length];
+    public int[] limelightTargets = new int[Limelight.values().length];
+    public boolean[] limelightSeesAprilTags = new boolean[Limelight.values().length];
+
+    public Pose2d[] limelightCalculatedPose = new Pose2d[Limelight.values().length];
+    public Pose2d limelightLastSeenPose = new Pose2d();
+    public double[] limelightAprilTagDistance = new double[Limelight.values().length];
+
+    public double[] limelightTimestamp = new double[Limelight.values().length];
+  }
+
+  default void updateInputs(VisionInputs inputs) {}
```

Comment:
I don't think we should use `default` in our interfaces. IMO it kind of goes against what interfaces should be, plus we shouldn't actually have any methods we don't implement.

Reply:
then open up a pr or an issue and change it

Reply:
tyty